policy/executor/internal/nodes.go

```diff
@@ -522,21 +522,32 @@ func (nodeData *ReportingJobNodeData) score() (*policy.Score, error) {
 		}
 	}
 
-	if nodeData.isQuery {
-		// if this reporting job represents a reporting query, we want to add that
-		// datapoints calculation, but return the score as is
-
-		if nodeData.childScores[nodeData.queryID] == nil {
-			panic("invalid reporting job")
-		}
+	if nodeData.forwardScore {
 		var s *policy.Score
-		if v := nodeData.childScores[nodeData.queryID].score; v == nil {
+
+		if len(nodeData.childScores) == 0 {
 			s = &policy.Score{
-				QrId: nodeData.queryID,
-				Type: policy.ScoreType_Result,
+				QrId:            nodeData.queryID,
+				Type:            policy.ScoreType_Unscored,
+				ScoreCompletion: 100,
 			}
 		} else {
-			s = proto.Clone(v).(*policy.Score)
+			if len(nodeData.childScores) != 1 {
+				panic("invalid reporting job")
+			}
+			var child string
+			for k := range nodeData.childScores {
+				child = k
+			}
+			if v := nodeData.childScores[child].score; v == nil {
+				s = &policy.Score{
+					QrId: nodeData.queryID,
+					Type: policy.ScoreType_Result,
+				}
+			} else {
+				s = proto.Clone(v).(*policy.Score)
```

Comment:
not sure how often this gets called but we had some issues with memory/cpu consumption a while back. A lot of those were caused by `proto.Clone`. It is doing a lot of allocations and reflection magic which is super inefficient. If we need to clone this structure often, it's best to manually write a function that does that

Reply:
i can change this in a follow up pr. Would like to avoid it being part of this one